cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/ZookeeperScalerTest.java

```diff
@@ -0,0 +1,301 @@
+        desired2.put("server.2", "127.0.0.1:28881:38881:participant;127.0.0.1:21811");
+
+        assertThat(ZookeeperScaler.isDifferent(current, desired2), is(true));
+
+        Map<String, String> desired3 = new HashMap<>(3);
+        desired3.put("server.2", "127.0.0.1:28881:38881:participant;127.0.0.1:21811");
+        desired3.put("server.3", "127.0.0.1:28882:38882:participant;127.0.0.1:21812");
+
+        assertThat(ZookeeperScaler.isDifferent(current, desired3), is(true));
+    }
+
+    @Test
+    public void testGenerateConfigOneNode() {
+        Map<String, String> expected = new HashMap<>(3);
+        expected.put("server.1", "127.0.0.1:28880:38880:participant;127.0.0.1:21810");
+
+        assertThat(ZookeeperScaler.generateConfig(1), is(expected));
+    }
+
+    @Test
+    public void testGenerateConfigThreeNodes() {
+        Map<String, String> expected = new HashMap<>(3);
+        expected.put("server.1", "127.0.0.1:28880:38880:participant;127.0.0.1:21810");
+        expected.put("server.2", "127.0.0.1:28881:38881:participant;127.0.0.1:21811");
+        expected.put("server.3", "127.0.0.1:28882:38882:participant;127.0.0.1:21812");
+
+        assertThat(ZookeeperScaler.generateConfig(3), is(expected));
+    }
+
+    @Test
+    public void testParseConfig() {
+        String config = "server.1=127.0.0.1:28880:38880:participant;127.0.0.1:21810\n" +
+                        "server.2=127.0.0.1:28881:38881:participant;127.0.0.1:21811\n" +
+                        "server.3=127.0.0.1:28882:38882:participant;127.0.0.1:21812\n" +
+                        "version=100000000b";
+
+        Map<String, String> expected = new HashMap<>(3);
+        expected.put("server.1", "127.0.0.1:28880:38880:participant;127.0.0.1:21810");
+        expected.put("server.2", "127.0.0.1:28881:38881:participant;127.0.0.1:21811");
+        expected.put("server.3", "127.0.0.1:28882:38882:participant;127.0.0.1:21812");
+
+        assertThat(ZookeeperScaler.parseConfig(config.getBytes(StandardCharsets.US_ASCII)), is(expected));
+    }
+
+    @Test
+    public void testMapToList() {
+        Map<String, String> servers = new HashMap<>(3);
+        servers.put("server.1", "127.0.0.1:28880:38880:participant;127.0.0.1:21810");
+        servers.put("server.2", "127.0.0.1:28881:38881:participant;127.0.0.1:21811");
+        servers.put("server.3", "127.0.0.1:28882:38882:participant;127.0.0.1:21812");
+
+        List<String> expected = new ArrayList<>(3);
+        expected.add("server.1=127.0.0.1:28880:38880:participant;127.0.0.1:21810");
+        expected.add("server.2=127.0.0.1:28881:38881:participant;127.0.0.1:21811");
+        expected.add("server.3=127.0.0.1:28882:38882:participant;127.0.0.1:21812");
+
+        assertThat(ZookeeperScaler.serversMapToList(servers), containsInAnyOrder(expected.toArray()));
+    }
+
+    @Test
+    public void testTimeoutingConnection(VertxTestContext context)  {
```

Comment:
Timeouting isn't a word in English (though I quite like it)

```suggestion
    public void testConnectionTimeout(VertxTestContext context)  {
```